src/main/java/org/prebid/server/handler/ForceDealsUpdateHandler.java

```diff
@@ -0,0 +1,119 @@
+
+    public ForceDealsUpdateHandler(DeliveryStatsService deliveryStatsService,
+                                   PlannerService plannerService,
+                                   RegisterService registerService,
+                                   AlertHttpService alertHttpService,
+                                   DeliveryProgressService deliveryProgressService,
+                                   LineItemService lineItemService,
+                                   String endpoint) {
+
+        this.deliveryStatsService = Objects.requireNonNull(deliveryStatsService);
+        this.plannerService = Objects.requireNonNull(plannerService);
+        this.registerService = Objects.requireNonNull(registerService);
+        this.alertHttpService = Objects.requireNonNull(alertHttpService);
+        this.deliveryProgressService = Objects.requireNonNull(deliveryProgressService);
+        this.lineItemService = Objects.requireNonNull(lineItemService);
+        this.endpoint = Objects.requireNonNull(endpoint);
+    }
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        try {
+            handleDealsAction(dealsActionFrom(routingContext));
+            HttpUtil.executeSafely(routingContext, endpoint,
+                    response -> response
+                            .setStatusCode(HttpResponseStatus.NO_CONTENT.code())
+                            .end());
+        } catch (InvalidRequestException e) {
+            respondWithError(routingContext, HttpResponseStatus.BAD_REQUEST, e);
+        } catch (Exception e) {
+            respondWithError(routingContext, HttpResponseStatus.INTERNAL_SERVER_ERROR, e);
+        }
+    }
+
+    private void handleDealsAction(DealsAction dealsAction) {
+        switch (dealsAction) {
+            case UPDATE_LINE_ITEMS:
+                plannerService.updateLineItemMetaData();
+                break;
+            case SEND_REPORT:
+                deliveryStatsService.sendDeliveryProgressReports();
+                break;
+            case REGISTER_INSTANCE:
+                registerService.performRegistration();
+                break;
+            case RESET_ALERT_COUNT:
+                alertHttpService.resetAlertCount("pbs-register-client-error");
+                alertHttpService.resetAlertCount("pbs-planner-client-error");
+                alertHttpService.resetAlertCount("pbs-planner-empty-response-error");
+                alertHttpService.resetAlertCount("pbs-delivery-stats-client-error");
+                break;
+            case CREATE_REPORT:
+                deliveryProgressService.createDeliveryProgressReports(ZonedDateTime.now());
+                break;
+            case INVALIDATE_LINE_ITEMS:
+                lineItemService.invalidateLineItems();
+                break;
+            default:
+                throw new IllegalStateException("Unexpected action value");
+        }
+    }
+
+    private DealsAction dealsActionFrom(RoutingContext routingContext) {
```

Comment:
1. Can be static.
2. Pls move this method declaration before `handleDealsAction` to comply method call ordering.